Document/0x04i-Testing-User-Privacy-Protection.md

```diff
@@ -0,0 +1,136 @@
+- **Transparency**:
+  - Users should be able to request all information that the application has on them, and be explained how to request this information.
+  - Includes: privacy policies, user education, proper logging and auditing mechanisms, etc.
+- **Intervenability**:
+  - Users should be able to correct their personal information, request its deletion and withdraw any given consent at any time, and be explained how to do so.
+  - Includes: privacy settings directly in the app, single points of contact for individuals’ intervention requests (e.g. in-app chat, telephone number, e-mail), etc.
+
+> See Section 5.1.1 "Introduction to data protection goals" in ENISA's ["Privacy and data protection in mobile applications"](https%3A%2F%2Fwww.enisa.europa.eu%2Fpublications%2Fprivacy-and-data-protection-in-mobile-applications%2Fat_download%2FfullReport&usg=AOvVaw06m90YDUaLCeeD2r-Ompgn) for more detailed descriptions.
+
+Addressing both security and privacy protection goals at the same time is a very challenging task (if not impossible in many cases). There's even an interesting visualization in IEEE's publication [Protection Goals for Privacy Engineering](https://ieeexplore.ieee.org/document/7163220) called ["The Three Axes"](https://ieeexplore.ieee.org/document/7163220#sec2e) representing the impossibility to ensure 100% of each of the six goals simultaneously.
+
+Most part of the processes derived from the protection goals are traditionally being covered in a privacy policy. However, this approach is not always optimal:
+
+- developers are not legal experts but still need to be compliant.
+- users would be require to read usually long and wordly policies.
+
+### The New Approach (Google's and Apple's take on this)
+
+In order to address these challenges and help users easily understand how their data is being collected, handled and shared, Google and Apple introduced new privacy labeling systems (very much along the lines of NIST's proposal for [Consumer Software Cybersecurity Labeling](https://www.nist.gov/system/files/documents/2021/11/01/Draft%20Consumer%20Software%20Labeling.pdf):
+
+- the App Store [Nutrition Labels](https://www.apple.com/privacy/labels/) (since 2020).
+- the Google Play [Data Safety Labels](https://android-developers.googleblog.com/2021/05/new-safety-section-in-google-play-will.html) (since 2021).
+
+As a new requirement on both platforms, it's vital that these labels are accurate in order to provide user assurance and mitigate developer abuse.
+
+### Common Violations that Can Be Addressed with the New Approach
+
+This is a non-exhaustive list of common violations that you as as security tester should report:
+
+- An app collects device location but does not have a prominent disclosure explaining which feature uses this data and/or indicates the app's usage in the background.
+- An app has a runtime permission requesting access to data before the prominent disclosure which specifies what the data is used for.
```

Comment:
```suggestion
- An app has a runtime permission requesting access to data before the prominent disclosure which specifies why the data is used.
```

Reply:
Yes, this one is just another example of slightly more formal language because "what for" and "why" are very interchangeable. It's better to not end a sentence on a preposition like "for," but it's completely your style choice because they do have the same meaning.

Reply:
ok, great! Thanks for the explanation :)